src/Cache/FileHandler.php

```diff
@@ -72,5 +81,7 @@ public function write(CacheInterface $cache)
                 $this->file
             );
         }
+
+        @chmod($this->file, 0666 & ~umask());
```

Comment:
why setting the permission of file after we already write to it ?

Reply:
When you run as `root` with very strict settings the `cache` file might be created with write permissions for `root` only. When you than rerun as a regular user the `cache` cannot be modified.

Reply:
but 666 gave permission to all users, root or not

Reply:
We never set the permissions to `0666` I think? Using `0666` with `& ~umask()` should give `0666` or lower when configured by user.

Reply:
I'm not sure what will be outcome of umask here TBH, never had to use it.
IMO 666 for cache is ok. it's only cache

Reply:
yeah, `0666` will work, I've updated the PR to it.